File: Java/src/main/java/com/gildedrose/AgedBrieItem.java
package com.gildedrose;

public class AgedBrieItem extends UpdatableItem {
    public AgedBrieItem(int sellIn, int quality) {
        super("Aged Brie", sellIn, quality);
    }
    @Override
    public void update() {
        sellIn--;
        if (quality < 50) quality++;
        if (sellIn < 0 && quality < 50) quality++;
    }
} 

File: php/src/Item/AgedBrieItem.php
<?php

declare(strict_types=1);

namespace GildedRose\Item;

use GildedRose\Item;

final class AgedBrieItem extends Item
{
    public function update()
    {
        $this->increaseQuality();
        $this->sellIn -= 1;
        if ($this->sellIn < 0) {
            $this->increaseQuality();
        }
    }
}